File: src/leetcode/java/NextGreaterElementI_496.java
package leetcode.java;

import java.util.HashMap;

public class NextGreaterElementI_496 {
  public int[] nextGreaterElement(int[] findNums, int[] nums) {
    HashMap<Integer, Integer> map = new HashMap<>();
    for (int i = 0; i < nums.length; i++) loop1: {
      for (int j = i + 1; j < nums.length; j++) {
        if (nums[j] > nums[i]) {
          map.put(nums[i], nums[j]);
          break loop1;
        }
      }
      map.put(nums[i], -1);
    }
    for (int i = 0; i < findNums.length; i++) findNums[i] = map.get(findNums[i]);
    return findNums;
  }
}

File: src/leetcode/python/NextGreaterElementI_496.py
class NextGreaterElementI_496(object):
    def nextGreaterElement(self, findNums, nums):
        """
        :type findNums: List[int]
        :type nums: List[int]
        :rtype: List[int]
        """
        m = {}
        l = len(nums)
        for i in range(l):
            for j in range(i+1, l):
                if nums[j] > nums[i]:
                    map[nums[i]] = nums[j]
                    break
        for i in range(len(findNums)):
            findNums[i] = m.get(findNums[i], -1)
        return findNums
